web/app/scripts/src/directives/form.es.js

```diff
@@ -39,11 +50,49 @@ module.directive('surveysForm', [
                             // been set.
                             if (Object.keys(scope.model).length === 0) {
                                 scope.model = FieldsetParser.parseFormToModel(structure);
+                                countQuestionsTotal();
                             }
                         });
                     }
                 });
 
+                // answers is an object containing objects
+                // object {
+                //     1 : {
+                //         2 : 0
+                //         3 : 0
+                //     }
+                // }
+                // answered is a number of type number
+                const countNumberOfAnsweredQuestions = function (answers) {
+                    let answered = 0;
+
+                    for (let groupKey in answers) {
+                        let answerGroup = answers[groupKey]
+                        for (let answerKey in answerGroup) {
+                            let answer = answerGroup[answerKey]
+                            if (answer > 0 || answer.length > 0) {
+                                answered++;
+                            }
+                        }
+                    }
+
+                    return answered;
+                };
+
+                const calculatePercentageComplete = function (completedQuestions) {
+                    const result = ((completedQuestions / totalQuestionCount) * 100)
+                    if (!isNaN(result)) {
+                        scope.percentageComplete = Math.round(result) + '%';
+                    }
```

Comment:
the only case where `result` would be `NaN` is whern `totalQuestionsCount` is `0`. So what I think is the better solution is to make `scope.percentageComplete = '100%`;` in that case:
```
if (totalQuestoinsCount === 0) {
    scope.percentageComplete = '100%';
} else {
 const result = ....;
...
}
```
```